lang_java/tree_sitter/tree_sitter_ast_java_grammar_n.ml

```diff
@@ -0,0 +1,695 @@
+| Inferred of inferred_parameters
+
+
+and inferred_parameters = identifier * ( identifier list)
+
+and ternary_expression = {
+  condition: _expression;
+  consequence: _expression;
+  alternative: _expression;
+}
+
+and unary_expression = string wrap
+
+and update_expression = string wrap
+ 
+and _primary =
+| Literal of _literal
+| Class of class_literal
+| This of this
+| Parenthesized of parenthesized_expression
+| Object of object_creation_expression
+| Field of field_access
+| Array of array_access
+| Method of method_invocation
+| Method1 of method_reference
+| Array1 of array_creation_expression
+
+
+and array_creation_expression = {
+  t_type: _simple_type;
+  t_dimensions: interm1;
+  value: array_initializer;
+}
+
+and interm1 = 
+| Dimensions1 of dimensions_expr list
+| Dimensions of dimensions option
+
+and dimensions_expr = ( _annotation list) * _expression
+
+and parenthesized_expression = _expression
+
+and class_literal =
+| Ambiguous of _ambiguous_name
+| Numeric of _numeric_type
+| Boolean of boolean_type
+| Void of void_type
+
+
+and object_creation_expression =
+| Unqualified of _unqualified_object_creation_expression
+| Ambiguous of _ambiguous_name * _unqualified_object_creation_expression
+| Primary of _primary * _unqualified_object_creation_expression
+
+
+and _unqualified_object_creation_expression = {
+  type_arguments: type_arguments;
+  t_type: _simple_type;
+  arguments: argument_list;
+  class_body: class_body option;
+}
+
+and field_access =
+| Interm2 of interm2
+| Interm3 of interm3
+| Interm4 of interm4
+
+and interm2 = {
+  t_object: _ambiguous_name;
+  field: this;
+}
+
+and interm3 = {
+  t_object: _primary * super;
+  field: identifier;
+}
+
+and interm4 = {
+  t_object: _ambiguous_name;
+  mods: super;
+  field: identifier;
+}
+  
+
+and array_access = {
+  array: _ambiguous_name * _primary;
+  index: _expression;
+}
+
+and method_invocation = 
+| Interm5 of interm5
+| Interm6 of interm6
+| Interm7 of interm7
+
+and interm5 = {
+  t_object: _ambiguous_name * _primary * super * _ambiguous_name;
+  type_arguments: type_arguments;
+  name: identifier;
+}
+
+and interm6 = {
+  t_object: _ambiguous_name;
+  supertype_arguments: type_arguments;
+  name: identifier;
+  arguments: argument_list;
+}
+
+and interm7 = {
+  name: identifier * _reserved_identifier;
+}
+
+and argument_list = (_expression * ( _expression list)) option
+
+and method_reference = interm8 * type_arguments option * identifier option
+
+and interm8 = 
+| Type of _type
+| Ambiguous of _ambiguous_name
+| Primary of _primary
+| Super of super
+
+and type_arguments = (interm9 * (_type * wildcard) list) option
+
+and interm9 =
+| Type of _type
+| Wildcard of wildcard
+
+and wildcard = ( _annotation list) * _wildcard_bounds option
+
+and _wildcard_bounds =
+| Type of _type
+| Super of super * _type
+
+and dimensions = ( _annotation list) 
+
+and _statement =
+| Declaration of _declaration
+| Expression of expression_statement
```

Comment:
You have already used 'Expression' as a constructor in another type, so this must be renamed
Expression1 (or Expression42, whatever is unique in the whole file)